jax/experimental/export/_export.py

```diff
@@ -391,71 +388,58 @@ def do_export(*args_specs, **kwargs_specs) -> Exported:
-      allow_non_replicated_sharding = True
 
     if lowering_platforms is not None:
       actual_lowering_platforms = tuple(lowering_platforms)
     else:
       actual_lowering_platforms = (default_lowering_platform(),)
 
-    # Do not include shape assertions if the version is < 7.
-    enable_shape_assertions = (
-        DisabledSafetyCheck.shape_assertions() not in disabled_checks and
-        version >= _VERSION_START_SUPPORT_SHAPE_ASSERTIONS)  # type: ignore
-    try:
-      prev_enable_shape_assertions = _shape_poly.thread_local_state.enable_shape_assertions
-      _shape_poly.thread_local_state.enable_shape_assertions = enable_shape_assertions
-      replace_tokens_with_dummy = not _keep_main_tokens(version)
-
-      symbolic_scope: tuple[_shape_poly.SymbolicScope, tree_util.KeyPath] | None = None
-      for k_path, aval in tree_util.tree_flatten_with_path((args_specs, kwargs_specs))[0]:
-        # Static args may has no `shape` attribute.
-        if not hasattr(aval, "shape"):
-          continue
-        for d in aval.shape:
-          if _shape_poly.is_symbolic_dim(d):
-            if symbolic_scope is None:
-              symbolic_scope = (d.scope, k_path)
-              continue
-            symbolic_scope[0]._check_same_scope(
-                d, when=f"when exporting {fun_name}",
-                self_descr=f"current (from {_shape_poly.args_kwargs_path_to_str(symbolic_scope[1])}) ",
-                other_descr=_shape_poly.args_kwargs_path_to_str(k_path))
-
-      lowered = wrapped_fun_jax.lower(
-          *args_specs, **kwargs_specs,
-          _experimental_lowering_parameters=mlir.LoweringParameters(
-            platforms=actual_lowering_platforms,
-            replace_tokens_with_dummy=replace_tokens_with_dummy,
-          ))
-
-      lowering = lowered._lowering  # type: ignore
-      _check_lowering(lowering)
-      mlir_module = lowering.stablehlo()
-
-      args_avals_flat, _ = tree_util.tree_flatten(lowered.in_avals)
-      if "mut" in lowering.compile_args:
-        if lowering.compile_args["mut"]: raise NotImplementedError
-      if "kept_var_idx" in lowering.compile_args:
-        module_kept_var_idx = tuple(sorted(lowering.compile_args["kept_var_idx"]))
-      else:
-        # For pmap
-        module_kept_var_idx = tuple(range(len(args_avals_flat)))
-      shape_poly_state = lowering.compile_args["shape_poly_state"]
-      if (not all(core.is_constant_shape(a.shape) for a in args_avals_flat)
-          or lowering.compile_args.get("ordered_effects", [])):
-        mlir_module = _wrap_main_func(
-            mlir_module, args_avals_flat, args_kwargs_tree=lowered.in_tree,
-            has_platform_index_argument=shape_poly_state.has_platform_index_argument,
-            module_kept_var_idx=module_kept_var_idx,
-            serialization_version=version)
-    finally:
-      _shape_poly.thread_local_state.enable_shape_assertions = prev_enable_shape_assertions
+    # TODO: move to `lower`
```

Comment:
Consider adding a few more details about what needs to be done.

Reply:
This is really in an upcoming PR, this TODO won't survive long